.github/workflows/release.yml

```diff
@@ -0,0 +1,125 @@
+# Generate binary releases when a tag is pushed
+name: release
+
+on:
+  push:
+    tags:
+      - 'v*.*.*'
+
+jobs:
+
+  create-windows-binaries:
+
+    runs-on: windows-latest
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Install stable
+      uses: dtolnay/rust-toolchain@stable
+      with:
+        toolchain: stable
+
+    - name: Build cargo-careful
+      run: |
+        cargo build --release
+
+    - name: Build package
+      id: package
+      shell: bash
+      run: |
+        echo "file=./target/release/cargo-careful.exe" >> $GITHUB_OUTPUT
+        echo "name=cargo-careful.exe" >> $GITHUB_OUTPUT
+
+    - name: Upload artifacts
+      uses: actions/upload-artifact@v2
+      with:
+        name: ${{ steps.package.outputs.name }}
+        path: ${{ steps.package.outputs.file }}
+
+  create-unix-binaries:
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        include:
+          - os: ubuntu-latest
+            target: x86_64-unknown-linux-musl
+          - os: macos-latest
+            target: x86_64-apple-darwin
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Install Rust stable
+      uses: dtolnay/rust-toolchain@stable
+      with:
+        toolchain: stable
+        target: ${{ matrix.target }}
+
+    - name: Install musl
+      if: contains(matrix.target, 'linux-musl')
+      run: |
+        sudo apt-get install musl-tools
+
+    - name: Build cargo-careful
+      run: |
+        RUSTFLAGS=+crt-static cargo build --release --target ${{ matrix.target }}
+
+    - name: Strip binary
+      run: |
+        strip target/${{ matrix.target }}/release/cargo-careful
+
+    - name: Sanity check version
+      run: |
+        VERSION=v$(cargo pkgid | cut -d# -f2)
+        [[ "$VERSION" == "${{ github.ref_name }}" ]] || \
+        echo "Cargo version $VERSION and tag ${{ github.ref_name}} differ" && exit 1
```

Comment:
This also doesn't work, needs parentheses around the `echo && exit` it seems.

In the future please either test what you submit as a PR, or state clearly that it has not been tested. Otherwise this creates a bunch of emergency work for maintainers.